cookbooks/bcpc-hadoop/recipes/hbase_config.rb

```diff
@@ -134,7 +134,8 @@
 # If HBASE bucket cache is enabled the properties from this section will be included in hbase-site.xml
 #
 bucketcache_size = (node['bcpc']['hadoop']['hbase_rs']['mx_dir_mem']['size'] - node['bcpc']['hadoop']['hbase_rs']['hdfs_dir_mem']['size']).floor
-if node['bcpc']['hadoop']['hbase']['bucketcache']['enabled'] == true
+if node['bcpc']['hadoop']['hbase']['bucketcache']['enabled'] == true &&
+   node['bcpc']['hadoop']['rs_hosts'].map { |rs| rs.values[0] }.include?(node['fqdn'])
```

Comment:
Using `include` may cause foo-r1n1 to match foo-r1n10. Why not a `select` over `map` and look for equality?

Reply:
I don't think that will happen because the result of a map is an Array. 
It will call the include? which falls back to the method in Enumerable.
That returns true if any element in the map results == node['fqdn']

Reply:
What Ron said. In the map, I strip the small hashes down to just the value. The result is a `String` array of just FQDNs. I then do an `include?` using the fqdn of the node itself to check for a match, which should avoid collisions.